Crypto/PubKey/Curve25519.hs

```diff
@@ -110,3 +119,22 @@ foreign import ccall "cryptonite_curve25519_donna"
                            -> Ptr Word8 -- ^ secret
                            -> Ptr Word8 -- ^ basepoint
                            -> IO ()
+
+generateSecretKey :: MonadRandom m => m SecretKey
+generateSecretKey = return $ unsafeDoIO $ do
```

Comment:
you really don't want to put things into a bytestring and then moved it to a Scrubbed bytes. it does defeat the purpose. getRandomBytes should gives you Scrubbed Bytes too if you want.